packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -564,37 +578,90 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         }
     }
 
-    private tooltip(x: number, y: number): void {
+    private closestPoint(array: any, point: any, xRatio: number, yRatio: number, chartHeight: number): number | undefined {
+        const hypos: any[] = [];
+
+        array.forEach((ar: any) => {
+            // Convert x values to screen pixels
+            const pointX = point.x * xRatio;
+            const x = ar.x * xRatio;
```

Comment:
When zoomed in the x values don't seem correct, e.g. a range of 40 px instead of 600 px for a time range of 180,000,000 ns.

Reply:
True. xRatio was being calculated with `offset` instead of `start`, which works only without zooming. Fixed.